pkg/core/stateroot/store.go

```diff
@@ -23,7 +23,14 @@ func (s *Module) addLocalStateRoot(sr *state.MPTRoot) error {
 
 	data := make([]byte, 4)
 	binary.LittleEndian.PutUint32(data, sr.Index)
-	return s.Store.Put([]byte{byte(storage.DataMPT), prefixLocal}, data)
+	if err := s.Store.Put([]byte{byte(storage.DataMPT), prefixLocal}, data); err != nil {
+		return err
+	}
+	s.localHeight.Store(sr.Index)
+	if s.signAndSendCb != nil {
+		return s.signAndSendCb(sr)
```

Comment:
We better do this asynchronously, notifying service via channel (some subscription mechanism?), that's kinda the point of SV design (and we don't want to do sends in `storeBlock`). Or actually the service can subscribe to new blocks, because there is no new block without stateroot.

Reply:
Fixed.